File: MyAnimeList/src/main/java/com/example/myanimelist/di/modules/AnimeRepositoryModule.java
package com.example.myanimelist.di.modules;

import com.example.myanimelist.repositories.animes.AnimeRepository;
import com.example.myanimelist.repositories.animes.IAnimeRepository;
import dagger.Binds;
import dagger.Module;

import javax.inject.Singleton;

@Module
public interface AnimeRepositoryModule {
    @Singleton
    @Binds
    IAnimeRepository bindAnimeRepository(AnimeRepository repository);
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/di/modules/AnimeRepositoryModule.kt
package com.example.myanimelist.di.modules

import com.example.myanimelist.repositories.animes.AnimeRepository
import com.example.myanimelist.repositories.animes.IAnimeRepository
import dagger.Binds
import dagger.Module
import javax.inject.Singleton

@Module
interface AnimeRepositoryModule {
@Singleton
@Binds
fun bindAnimeRepository(repository: AnimeRepository) : IAnimeRepository
}